fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/eventmappings/AllNodeEventsOneEvent.java

```diff
@@ -49,6 +56,30 @@
 
     private final static Logger LOGGER = getLogger(AllNodeEventsOneEvent.class);
 
+    /**
+     * If the only event is a modification of the jcr:lastModified it is most likely the effect of an inbound reference.
+     * Wrap the old event in a new one that replaces the event type with FEDORA_JCR_REFERENCE for later mapping.
+     *
+     * @param ev The list of events to check.
+     * @return The original list or a list with the altered event.
+     */
+    private static final List<Event> AlterReferenceEvents(final List<Event> ev) {
+        try {
+            if (ev.size() == 1 && ev.get(0).getPath().endsWith("/" + JCR_LASTMODIFIED) &&
+                ev.get(0).getType() == PROPERTY_CHANGED) {
+                final Event original = ev.get(0);
+                final Event tempEv =
+                    new WrappedJcrEvent((org.modeshape.jcr.api.observation.Event) original, FEDORA_JCR_REFERENCE);
+                final List<Event> list = new ArrayList<Event>();
```

Comment:
Is this a functionally equivalent definition of the `List` in Java 8? Just wondering when I should or should not add the type to the instantiation.

Reply:
It's just syntactic sugar, so yes.  Specifying `<Event>` twice is redundant; the compiler can infer that `new ArrayList<>` really means `new ArrayList<Event>` because you already said it's a `List<Event>`